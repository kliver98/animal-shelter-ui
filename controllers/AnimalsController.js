import axios from 'axios';

export const AnimalController = {
    register(animal) {
        return axios({
            method: 'POST',
            baseURL: process.env.API,
            url: `animals`,
            data: animal,
        })
    },
    list() {
        return axios({
            method: 'GET',
            baseURL: process.env.API,
            url: 'animals'
        });
    },
    delete(name) {
        return axios({
            method: 'DELETE',
            baseURL: process.env.API,
            url: `animals/${name}`,
        });
    },
    getAnimal(name) {
        return axios({
            method: 'GET',
            baseURL: process.env.API,
            url: `animals/${name}`,
        });
    },
    updateAnimal(name) {
        return axios({
            method: 'PUT',
            baseURL: process.env.API,
            url: `animals/${name}`,
        });
    }
}

