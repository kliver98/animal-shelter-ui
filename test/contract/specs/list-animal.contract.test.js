import { provider } from './init-pact';
import { AnimalController } from '../../../controllers';
import { Matchers } from '@pact-foundation/pact';

describe('Given an animal service', () => {
    beforeAll(async() => {
        await provider.setup();
    });

    describe('When a request to list all the animal is made', () => {
        beforeAll(async() => {
            await provider.addInteraction({
                state: 'has animals',
                uponReceiving: 'a request to list all animals',
                withRequest: {
                    method: 'GET',
                    path: '/animals'
                },
                willRespondWith: {
                    status: 200,
                    body: Matchers.eachLike({
                        name: Matchers.string('Manchas'),
                        breed: Matchers.like("Bengali"),
                        gender: Matchers.like("Female"),
                        vaccinated: Matchers.boolean(true),
                    }, {min: 1})
                }
            });
        });

        test('Then it should return the right data', async() => {
            const response = await AnimalController.list();
            expect(response.data).toMatchSnapshot();

            await provider.verify();
        })

    });

    afterAll(async () => {
        await provider.finalize();
    });
});
