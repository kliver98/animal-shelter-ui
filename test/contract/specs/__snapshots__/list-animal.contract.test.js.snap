// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given an animal sevrice Where a request to list all the animal is made Then it should return the right data 1`] = `
Array [
  Object {
    "breed": "Bengali",
    "gender": "Female",
    "isVaccinated": true,
    "name": "Manchas",
    "vaccines": Array [
      "rabia",
    ],
  },
]
`;
